### ISSUE TITLE
New module type - lookup modules

### ISSUE THREAD

**Issue by iglocska** [MISP/misp-modules#112]
Would simply tie into a search system in MISP

- do grep queries against a directory of files (or a subset of them)
  - for example, do a search for IP addresses in all PCAP files visilbe to the user
  - find all attributes that are PCAP files visible to the user
  - pass to search module
  - search module parses the files / indeces from the attachment file store
  - returns valid IDs with the values contained

**Comment by adulau** [comment]
So it would be a kind of grep-misp using the attributes as query and get back the matching events?

**Comment by iglocska** [comment]
Yep. Could be grep, could be other lookups. Any external lookup - like the discussion we had about known good hashes, etc. 

Basically the idea would be to take a user search input and a module name, do the magic depending on the module type and return a list of hits based on the results.

Hits could be either references to external resources or MISP-internal resources.
